src/services/RedisService.ts

```diff
@@ -0,0 +1,24 @@
+import { createClient, RedisClient } from 'redis';
+import { injectable } from 'inversify';
+import { promisify } from 'util';
+
+export const getAsync = (client: RedisClient) => promisify(client.get).bind(client);
+export const setAsync = (client: RedisClient) => promisify(client.set).bind(client);
+
+@injectable()
+export class RedisService {
+
+    public readonly client: RedisClient;
+    constructor() {
+        this.client = this.createClient();
+    }
+
+    private createClient(): RedisClient {
```

Comment:
I don't feel the redis client configuration belongs here in the service layer, we could create the client in the main server.js or somewhere similar? Also, could we inject the host/port properties from a config file too?

Reply:
Agreed, we could make the RedisClient injectable as there is ever only going to be one instance of it?

Reply:
Yes we could! Just seen an example here -> https://github.com/inversify/inversify-express-example/tree/master/BindingDecorators/utils/mongodb

Reply:
and then we could inject the client like here -> https://github.com/inversify/inversify-express-example/blob/master/BindingDecorators/service/user.ts